test/should_fail/exhaustive_float.erl

```diff
@@ -0,0 +1,12 @@
+-module(exhaustive_float).
+
+-export([ef/1]).
+
+-type t() :: {int, integer()}
+           | {float, float()}.
+
+-spec ef(t()) -> ok.
+ef(T) ->
+    case T of
+        {int, _} -> ok
+    end.
```

Comment:
Indeed, I think the direction is to expand exhaustiveness checking to as many basic types as possible for the widest coverage and the best level of feedback. For example, for real-world usability `binary()` should also be handled, since that's the actual type of string in Elixir and many Erlang projects, too.

We won't crash on `pick_value` for binaries yet, because `refinable` gates what might ever reach `pick_value` - so if `refinable(T) == false`, then `pick_value(T)` does not have to be defined.

Anyway, I would do it _gradually_ :) Adding some types to `refinable`/`pick_value` is trivial, but with some it's not the case.